### ISSUE TITLE
Code is not getting Update in IOS on TestFlight build 

### ISSUE THREAD

**Issue by Malau172** [microsoft/react-native-code-push#2643]
Thanks so much for filing an issue or feature request! Please fill out the following (wherever relevant):

### Steps to Reproduce
1.I'm uploading a codepush to the staging environment, my codepush doesn't show any errors in the terminal, but the changes don't reflect in the iOS app in testflght

Dependecies: - CodePush (from `../node_modules/react-native-code-push`)

**Comment by eramudeep** [comment]
> Thanks so much for filing an issue or feature request! Please fill out the following (wherever relevant):
> 
> ### Steps to Reproduce
> 1.I'm uploading a codepush to the staging environment, my codepush doesn't show any errors in the terminal, but the changes don't reflect in the iOS app in testflght
> 
> Dependecies: - CodePush (from `../node_modules/react-native-code-push`)

i am also facing the same. i Tried on Staging and Prod.

**Comment by Malau172** [comment]
@eramudeep Hi, did you manage to solve your problem with codepush?

**Comment by MikhailSuendukov** [comment]
Hello and thanks for reaching out to us, could you please try the solution from this [duplicate](https://github.com/microsoft/react-native-code-push/issues/1714) along with the latest version of react-native-code-push (8.2.1) and let us know if this solved the problem?

**Comment by hotaryuzaki** [comment]
same with me here, my scenario is:
1. installed to physical device using IPA or test flight
2. publish new update
3. successfully installed when we check in published history
4. the change not implemented at all

there is no issue in the debug because we got this in our log
```
 LOG  [CodePush] Installing update.
 LOG  [CodePush] Update is installed and will be run on the next app restart.
...
 LOG  [CodePush] Checking for update.
 LOG  [CodePush] Reporting CodePush update success (v4)
```

**Comment by rizshivalli** [comment]
+1

**Comment by hotaryuzaki** [comment]
> same with me here, my scenario is:
> 
> 1. installed to physical device using IPA or test flight
> 2. publish new update
> 3. successfully installed when we check in published history
> 4. the change not implemented at all
> 
> there is no issue in the debug simulator because we got this in our log
> 
> ```
>  LOG  [CodePush] Installing update.
>  LOG  [CodePush] Update is installed and will be run on the next app restart.
> ...
>  LOG  [CodePush] Checking for update.
>  LOG  [CodePush] Reporting CodePush update success (v4)
> ```

we just facing this issue in iOS

react-native: 0.71.14
react-native-code-push: 8.1.0
CocoaPods: 1.14.3
Xcode: 15.1/15C65

**Comment by Jonnboy91** [comment]
Having the same issue with:
"react-native": "0.73.4",
"react-native-code-push": "^8.2.1".

Tried both staging and production development, with mandatory updates, but just doesn't seem to update via Testflight. Tried with IPA as well via Xcode install after building it, but just doesn't seem to update at all currently. Just implemented this and working with Android without issues, but can't seem to get it working with iOS.

**Comment by hotaryuzaki** [comment]
> Having the same issue with: "react-native": "0.73.4", "react-native-code-push": "^8.2.1".
> 
> Tried both staging and production development, with mandatory updates, but just doesn't seem to update via Testflight. Tried with IPA as well via Xcode install after building it, but just doesn't seem to update at all currently. Just implemented this and working with Android without issues, but can't seem to get it working with iOS.

do you check the history in CLI?
i think you can check for staging apps to see if they get the update or not.
for me, the app gets the update, and the status is installed. but the update was not implemented.

**Comment by Jonnboy91** [comment]
> > Having the same issue with: "react-native": "0.73.4", "react-native-code-push": "^8.2.1".
> > Tried both staging and production development, with mandatory updates, but just doesn't seem to update via Testflight. Tried with IPA as well via Xcode install after building it, but just doesn't seem to update at all currently. Just implemented this and working with Android without issues, but can't seem to get it working with iOS.
> 
> do you check the history in CLI? i think you can check for staging apps to see if they get the update or not. for me, the app gets the update, and the status is installed. but the update was not implemented.

Hi, I'm using production now and was actually able to get it working on installing the IPA via Xcode after all and then running new updates to codepush, but the TestFlight at least when I last tried yesterday it did not work. I only shortly tested the staging and now working on it now, so forgot to check the CLI on that.

**Comment by theDeji** [comment]
Has anyone been able to find a fix for this ?

**Comment by Jonnboy91** [comment]
> Has anyone been able to find a fix for this ?

Have you tried production deployment key and when pushing the version to CodePush add the mandatory parameter?

That seemed to do the trick for me.

**Comment by talha-psychplus** [comment]
tried mandatory parameter but still not working.

**Comment by amanthegreatone** [comment]
pls check this i had this issue which was solved. https://github.com/microsoft/react-native-code-push/issues/2529#issuecomment-1968788585

**Comment by anhnguyen123** [comment]
> vui lòng kiểm tra cái này tôi đã giải quyết được vấn đề này. [#2529 (bình luận)](https://github.com/microsoft/react-native-code-push/issues/2529#issuecomment-1968788585)

My situation is like this now, what should I do?

![image](https://github.com/microsoft/react-native-code-push/assets/18522542/48a39b52-a24f-40cd-a87f-b91a2b8dcee5)

**Comment by sufian1999** [comment]
"react": "17.0.2",
"react-native": "0.68.2",
"react-native-code-push": "^7.1.0"

I pushed code with this:
appcenter codepush release-react -a owner-email/app-name -d Production -m --disabled

When the 'Just Now' changes to the Date Time. I Enabled the release and when I open the TestFlight, it downloads the update after 1-2 minutes and changes reflect on the next restart.

Today, I tested it 2 times on my TestFlight, it's working.

Suppose any issue comes up once my app is uploaded via the App Store. I will let you people know.
